src/pybind/mgr/snap_schedule/fs/schedule_client.py

```diff
@@ -111,10 +111,11 @@ def get_prune_set(candidates: Set[Tuple[cephfs.DirEntry, datetime]],
                         log.debug(('found enough snapshots for '
                                    f'{period_count}{period}'))
                         break
-    if len(keep) > MAX_SNAPS_PER_PATH:
-        log.info((f'Would keep more then {MAX_SNAPS_PER_PATH}, '
+    max_snaps_to_retain = retention.get("n", MAX_SNAPS_PER_PATH)
+    if len(keep) > max_snaps_to_retain:
+        log.info((f'Would keep more then {max_snaps_to_retain}, '
```

Comment:
If the number of `keep` is more than `max_snaps_to_retain` then we would keep them i.e. retain them, So if `keep` is 80 while `max_snaps_to_retain` is mentioned as 75, then log info would say `Would keep more then 75, pruning keep set`, this sounds as if we are defying the limit? Although the below code does answer it and the functionality does work fine but maybe this line can be made more meaningful?

Reply:
how would you like to have it phrased ?

Reply:
Ah, my bad for such a late response. Maybe this could be
```
log.info(f'Pruning keep set, would retain first {max_snaps_to_retain}'
         f' out of {keep} snaps')
```
what do you think?

Reply:
@dparmar18 `{keep}` would dump the entire list not just the keep length

Reply:
right then let's do `len(keep)`

Reply:
added a new commit to address this issue